feat(parser): report empty expression in JSX attribute error

leaysgur opened the pull request:
Fixes #16371

> TS17000: JSX attributes must only be assigned a non-empty 'expression'.

https://www.typescriptlang.org/dev/bug-workbench/?ignoreDeprecations=6#code/PTAEAEEsHMDsHsBOBTAIsgDigxgQwC6TywDOAXKAGwB0ADAFCQC2GS+oASsrtuwGaJ4TUAHIUPfCMYs2oAMKgBQ0dhEBuevQA8CjAF4A3sABUoACoB5VBdDHgAX1DAAfEA

Coverage results are unchanged:

https://github.com/oxc-project/oxc/blob/d490daa113c582703afef553c6537f592cb02a45/tasks/coverage/snapshots/parser_typescript.snap#L7295-L7317

Boshen commented:
> I found a real case for this https://github.com/coveo/plasma/blob/08ea8c6f4f1f53bb9d48874b17ceeeb991e22be6/packages/figma/src/Collection.figma.tsx

It's fine, those files are template files, which aren't meant to be formatted.

leaysgur committed:
feat(parser): report empty expression in JSX attribute error (#16378)

Fixes #16371

> TS17000: JSX attributes must only be assigned a non-empty 'expression'.

https://www.typescriptlang.org/dev/bug-workbench/?ignoreDeprecations=6#code/PTAEAEEsHMDsHsBOBTAIsgDigxgQwC6TywDOAXKAGwB0ADAFCQC2GS+oASsrtuwGaJ4TUAHIUPfCMYs2oAMKgBQ0dhEBuevQA8CjAF4A3sABUoACoB5VBdDHgAX1DAAfEA

Coverage results are unchanged:

https://github.com/oxc-project/oxc/blob/d490daa113c582703afef553c6537f592cb02a45/tasks/coverage/snapshots/parser_typescript.snap#L7295-L7317
